server/src/main/scala/org/http4s/server/staticcontent/FileService.scala

```diff
@@ -160,14 +169,15 @@ object FileService {
               }
               .value
           }
-        else
-          F.delay(file.length()).map { size =>
-            Some(
-              Response[F](
-                status = Status.RangeNotSatisfiable,
-                headers = v2.Headers
-                  .apply(AcceptRangeHeader, `Content-Range`(SubRange(0, size - 1), Some(size)))))
-          }
-      case _ => F.pure(None)
+          else nope
+      case _ =>
+        req.headers.get(CIString("Range")) match {
```

Comment:
This is where I'm thinking the Select machinery returning an `Ior` might carry some weight.  We have to handle headers that just vanish because they're invalid.